src/components/ChildHideCheckbox.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { useContext } from 'react';
+import { Checkbox, Tooltip } from 'antd';
+
+import { ChildCheckboxProps } from '../types';
+import { VisibilityContext } from '../AgentVisibilityContext';
+import { CheckboxState, tooltipMap } from '../constants';
+
+const ChildHideCheckbox: React.FunctionComponent<ChildCheckboxProps> = (
+  props: ChildCheckboxProps
+): JSX.Element => {
+  const { name, parentName } = props;
+  const { handleHideChildCheckboxChange, hiddenAgents } =
+    useContext(VisibilityContext);
+
+  const selections = hiddenAgents[parentName];
+
+  const getCheckboxStatus = () => {
+    if (selections?.includes(name)) {
```

Comment:
I asked this before, but why is selections possibly undefined? shouldn't it always be an array?

Reply:
Selections is specific to the agent, so in the initial state before the UI display data arrives this would be undefined.